src/crimson/os/seastore/extentmap_manager/btree/btree_extentmap_manager.h

```diff
@@ -0,0 +1,73 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <iostream>
+
+#include <boost/intrusive_ptr.hpp>
+#include <boost/smart_ptr/intrusive_ref_counter.hpp>
+#include <seastar/core/future.hh>
+
+#include "include/ceph_assert.h"
+#include "include/buffer_fwd.h"
+#include "crimson/osd/exceptions.h"
+
+#include "crimson/os/seastore/extentmap_manager.h"
+#include "crimson/os/seastore/extentmap_manager/btree/extentmap_btree_node.h"
+#include "crimson/os/seastore/seastore_types.h"
+#include "crimson/os/seastore/transaction_manager.h"
+
+namespace crimson::os::seastore::extentmap_manager {
+/**
+ * BtreeExtentMapManager
+ *
+ * Uses a btree to track :
+ * objaddr_t -> laddr_t mapping for each onode extentmap
+ *
+ */
+
+class BtreeExtentMapManager : public ExtentMapManager {
+  TransactionManager *tm = nullptr;
+  extmap_root_ref extmap_root = nullptr;
+
+  using get_root_ertr = TransactionManager::read_extent_ertr;
+  using get_root_ret = get_root_ertr::future<ExtMapNodeRef>;
+  get_root_ret get_extmap_root(Transaction &t);
+
+  using insert_lextent_ertr = TransactionManager::read_extent_ertr;
+  using insert_lextent_ret = insert_lextent_ertr::future<ExtentRef>;
+  insert_lextent_ret insert_lextent(Transaction &t,
+                                    ExtMapNodeRef extent, objaddr_t lo,
+				    lext_map_val_t val);
+
+public:
+  explicit BtreeExtentMapManager(
+    TransactionManager *tm, extmap_root_ref extmap_root);
+
+  explicit BtreeExtentMapManager(TransactionManager *tm);
+
+  //allocate extent map tree root node
```

Comment:
These methods are simply redeclarations of the methods in ExtentMapManager.  If you ensure that the comments there are sufficient, you shouldn't need to duplicate them here (the behavior must conform to the interface, right?)